### ISSUE TITLE
Registration getting failed in frontend

### ISSUE THREAD

**Issue by chiranjivSharma** [getmaxun/maxun#128]
After I downloaded the github folder and done with npm run start,I can open frontend in http://localhost:5173. But if I neter email and password and then click on Register,it shows error message Registration Failed. Please try again.
Could you please help with that.

**Comment by amhsirak** [comment]
@chiranjivSharma this is a common error. Share your logs.
Have you set environment variables in .env? Has database connected successfully?

**Comment by chiranjivSharma** [comment]
npm run start

> maxun@0.0.1 start
> concurrently -k "npm run server" "npm run client"

[1] 
[1] > maxun@0.0.1 client
[1] > vite
[1] 
[0] 
[0] > maxun@0.0.1 server
[0] > cross-env NODE_OPTIONS='--max-old-space-size=8000' nodemon server/src/server.ts
[0] 
[0] [nodemon] 2.0.22
[0] [nodemon] to restart at any time, enter `rs`
[0] [nodemon] watching path(s): *.*
[0] [nodemon] watching extensions: ts,json
[0] [nodemon] starting `ts-node server/src/server.ts`
[1] The CJS build of Vite's Node API is deprecated. See https://vite.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.
[1] NODE_ENV=production is not supported in the .env file. Only NODE_ENV=development is supported to create a development build of your project. If you need to set process.env.NODE_ENV, you can set it in the Vite config instead.
[1] 
[1]   VITE v5.4.10  ready in 713 ms
[1] 
[1]   ➜  Local:   http://localhost:5173/
[1]   ➜  Network: use --host to expose
[0] Environment variables: { REDIS_HOST: 'localhost', REDIS_PORT: '6379' }
[0] Worker is running...
[0] Connected to Redis!
[0] Bucket does not exist, but MinIO was connected.
[0] Jobs: {
[0]   active: 0,
[0]   completed: 0,
[0]   delayed: 0,
[0]   failed: 0,
[0]   paused: 0,
[0]   prioritized: 0,
[0]   waiting: 0,
[0]   'waiting-children': 0
[0] }
[0] Database connected successfully
[0] Database synced successfully!
[0] 2024-11-05T08:01:45.668Z info: Server listening on port 8080
 above is the output of npm run start

and below is my .env file content inside root folder

# App Setup
NODE_ENV=production                     # Set to 'development' or 'production' as required
JWT_SECRET=your_jwt_secret_key          # Replace with a secure JWT secret key
DB_NAME=maxun                           # Your PostgreSQL database name
DB_USER=postgres                        # PostgreSQL username
DB_PASSWORD=postgres                    # PostgreSQL password
DB_HOST=localhost                        # Host for PostgreSQL in Docker
DB_PORT=5432                            # Port for PostgreSQL (default: 5432)
ENCRYPTION_KEY=your_encryption_key      # Key for encrypting sensitive data (passwords and proxies)
MINIO_ENDPOINT=localhost                    # MinIO endpoint in Docker
MINIO_PORT=9000                         # Port for MinIO (default: 9000)
MINIO_ACCESS_KEY=minio       # MinIO access key
MINIO_SECRET_KEY=minio123       # MinIO secret key
REDIS_HOST=localhost                        # Redis host in Docker
REDIS_PORT=6379                         # Redis port (default: 6379)

# Backend URLs
BACKEND_URL=http://localhost:8080       # Internal URL for backend service
VITE_BACKEND_URL=http://localhost:8080  # URL used by frontend to connect to backend

# Optional Google OAuth settings for Google Sheet Integration
GOOGLE_CLIENT_ID=your_google_client_id
GOOGLE_CLIENT_SECRET=your_google_client_secret
GOOGLE_REDIRECT_URI=your_google_redirect_uri

# Telemetry Settings - Please keep it enabled. Keeping it enabled helps us understand how the product is used and assess the impact of any new changes. 
MAXUN_TELEMETRY=true

**Comment by amhsirak** [comment]
@chiranjivSharma share your browser console logs. Is it making a request to undefined/auth/register?

**Comment by chiranjivSharma** [comment]
Register.tsx:35 
        
        
       POST http://localhost:5173/undefined/auth/register 404 (Not Found)
dispatchXhrRequest @ axios.js?v=912d5e2f:702
xhrAdapter @ axios.js?v=912d5e2f:572
dispatchRequest @ axios.js?v=912d5e2f:879
request @ axios.js?v=912d5e2f:1135
Axios.<computed> @ axios.js?v=912d5e2f:1159
wrap @ axios.js?v=912d5e2f:15
submitForm @ Register.tsx:35
callCallback2 @ chunk-GKJBSOWT.js?v=912d5e2f:3674
invokeGuardedCallbackDev @ chunk-GKJBSOWT.js?v=912d5e2f:3699
invokeGuardedCallback @ chunk-GKJBSOWT.js?v=912d5e2f:3733
invokeGuardedCallbackAndCatchFirstError @ chunk-GKJBSOWT.js?v=912d5e2f:3736
executeDispatch @ chunk-GKJBSOWT.js?v=912d5e2f:7014
processDispatchQueueItemsInOrder @ chunk-GKJBSOWT.js?v=912d5e2f:7034
processDispatchQueue @ chunk-GKJBSOWT.js?v=912d5e2f:7043
dispatchEventsForPlugins @ chunk-GKJBSOWT.js?v=912d5e2f:7051
(anonymous) @ chunk-GKJBSOWT.js?v=912d5e2f:7174
batchedUpdates$1 @ chunk-GKJBSOWT.js?v=912d5e2f:18913
batchedUpdates @ chunk-GKJBSOWT.js?v=912d5e2f:3579
dispatchEventForPluginEventSystem @ chunk-GKJBSOWT.js?v=912d5e2f:7173
dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay @ chunk-GKJBSOWT.js?v=912d5e2f:5478
dispatchEvent @ chunk-GKJBSOWT.js?v=912d5e2f:5472
dispatchDiscreteEvent @ chunk-GKJBSOWT.js?v=912d5e2f:5449
Show 21 more frames
Show less


in browser console this error is coming

**Comment by amhsirak** [comment]
@chiranjivSharma yes, as I said it's making a request to undefined/auth/register. Have you pulled the latest code from master branch? we've tried fixing this error here: #121 

is this fix available in your codebase?

**Comment by chiranjivSharma** [comment]
Okay with latest version registration is successful. thank you
